packages/spiderfier/directives/marker-spider.ts

```diff
@@ -0,0 +1,227 @@
+
+import {LegColorOptions, SpiderOptions} from '../services/google-spider-types';
+
+/**
+ * AgmMarkerSpider spiderfies map marker if they are near together
+ *
+ * ### Example
+ * ```typescript
+ * import { Component } from '@angular/core';
+ *
+ * @Component({
+ *  selector: 'my-map-cmp',
+ *  styles: [`
+ *    agm-map {
+ *      height: 300px;
+ *    }
+ * `],
+ *  template: `
+ *    <agm-map [latitude]="lat" [longitude]="lng" [zoom]="zoom">
+ *      <agm-marker-spider>
+ *        <agm-marker [latitude]="lat" [longitude]="lng" [label]="'M'">
+ *        </agm-marker>
+ *        <agm-marker [latitude]="lat2" [longitude]="lng2" [label]="'N'">
+ *        </agm-marker>
+ *      </agm-marker-spider>
+ *    </agm-map>
+ *  `
+ * })
+ * ```
+ */
+@Directive({
+  selector: 'agm-marker-spider',
+  providers: [
+    SpiderManager,
+    {provide: MarkerManager, useExisting: SpiderManager},
+    InfoWindowManager,
+  ]
+})
+export class AgmMarkerSpider implements OnDestroy, OnChanges, OnInit, SpiderOptions {
+  /**
+   * The colors of the legs
+   */
+  @Input() legColors: LegColorOptions;
+
+  /**
+   * Disable listening for marker move events (in case markers won't be dragged/moved)
+   */
+  @Input() markersWontMove: boolean;
+
+  /**
+   * Disable listening for visibility events (in case markers won't be hidden)
+   */
+  @Input() markersWontHide: boolean;
+
+  /**
+   * Disable the OverlappingMarkerSpiderfier.markerStatus.SPIDERFIABLE event
+   */
+  @Input() basicFormatEvents: boolean;
+
+  /**
+   * Do not unspiderify the markers when one of them is clicked
+   */
+  @Input() keepSpiderfied: boolean;
+
+  /**
+   * Do not unspiderify when the map is clicked
+   */
+  @Input() ignoreMapClick: boolean;
+
+  /**
+   * The pixel radius within which a marker is considered overlapping
+   */
+  @Input() nearbyDistance: number;
+
+  /**
+   * The amount of markers before the spiderfier switches to spiral mode
+   */
+  @Input() circleSpiralSwitchover: number;
+
+  /**
+   * The distance for the spiderfied markers from the center (circle mode)
+   */
+  @Input() circleFootSeparation: number;
+
+  /**
+   * The starting angle for the spiderfied markers (circle mode)
+   */
+  @Input() circleStartAngle: number;
+
+  /**
+   * The distance for the spiderfied markers from the center (spiral mode)
+   */
+  @Input() spiralFootSeparation: number;
+
+  /**
+   * The distance between the spiderfied markers from the first to the second (spiral mode)
+   */
+  @Input() spiralLengthStart: number;
+
+  /**
+   * The increment to the distance for each consecutive marker (spiral mode)
+   */
+  @Input() spiralLengthFactor: number;
+
+  /**
+   * The thickness of the marker lines
+   */
+  @Input() legWeight: number;
+
+  /**
+   * Triggers when a marker is formatted, can be used to style the icon
+   */
+  @Output() format: EventEmitter<{ marker: Marker, status: string }> = new EventEmitter<{ marker: Marker, status: string }>();
```

Comment:
Can we extract this as a type? So others can use this in their code.

Reply:
Updated